CMakeLists.txt

```diff
@@ -17,38 +17,88 @@ message("-- Version: ${VERSION}")
 set(GENERIC_LIB_VERSION ${VERSION})
 string(SUBSTRING ${VERSION} 0 1 GENERIC_LIB_SOVERSION)
 
-# Try and enable C++11. Don't use C++14 because it doesn't work in some
-# configurations.
+# Import our CMake modules
 include(CheckCXXCompilerFlag)
 include(AddCXXCompilerFlag)
 include(CXXFeatureCheck)
 
-check_cxx_compiler_flag(-std=c++11 HAVE_FLAG_CXX_11)
-check_cxx_compiler_flag(-std=c++0x HAVE_FLAG_CXX_0X)
-if (HAVE_FLAG_CXX_11)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-elseif (HAVE_FLAG_CXX_0X)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+# Determine the C++ standard to use
+# The C and C++ standards to use
+set(BENCHMARK_CXX_STANDARD auto CACHE STRING
+  "The C++ standard to use; auto uses the latest supported by the compiler")
+set_property(CACHE BENCHMARK_CXX_STANDARD PROPERTY STRINGS
+  auto c++14 c++11 c++98)
+if ("${BENCHMARK_CXX_STANDARD}" STREQUAL "auto")
+  # Try and enable C++11. Don't use C++14 because it doesn't work in some
+  # configurations.
+  #add_cxx_compiler_flag(-std=c++14)
+  if (NOT HAVE_CXX_FLAG_STD_CXX14)
+    add_cxx_compiler_flag(-std=c++11)
+    if (NOT HAVE_CXX_FLAG_STD_CXX11)
+      add_cxx_compiler_flag(-std=c++98)
+    endif()
+  endif()
+else()
+  add_cxx_compiler_flag(-std=${BENCHMARK_CXX_STANDARD})
 endif()
 
 # Turn compiler warnings up to 11
 add_cxx_compiler_flag(-Wall)
 add_cxx_compiler_flag(-Wextra)
 add_cxx_compiler_flag(-Wshadow)
-add_cxx_compiler_flag(-Werror)
-add_cxx_compiler_flag(-pedantic-errors)
+add_cxx_compiler_flag(-Werror RELEASE)
+add_cxx_compiler_flag(-pedantic)
+add_cxx_compiler_flag(-pedantic-errors RELEASE)
 add_cxx_compiler_flag(-Wshorten-64-to-32)
 add_cxx_compiler_flag(-Wfloat-equal)
 add_cxx_compiler_flag(-Wzero-as-null-pointer-constant)
-# Release flags
-add_cxx_compiler_flag(-fno-strict-aliasing RELEASE)
-
 add_cxx_compiler_flag(-Wthread-safety)
 if (HAVE_WTHREAD_SAFETY)
   add_definitions(-DHAVE_WTHREAD_SAFETY)
   cxx_feature_check(THREAD_SAFETY_ATTRIBUTES)
 endif()
 
+# Debug flags
+add_cxx_compiler_flag(-fsanitize=address DEBUG)
```

Comment:
If you are using clang as your c++ compiler it will require a sanitized standard library. Not sure about GCC. Can you pull these changes out for now? They don't seem to fit.